### PR TITLE
Missions

### PR DESCRIPTION
**Pull request by matthewfernst**
mission

### CODE REVIEW

**Review by mrrosoff — DISMISSED** [review]

File: client/components/App/HeaderBar/Missions.jsx [hunk 1]
```diff
@@ -1,55 +1,36 @@
...
-import { useMutation, useQuery } from "@apollo/client";
+import { useQuery } from "@apollo/client";
 import { GetHeaderNotifications } from "../../../graphql/query.js";
-import { ResolveInviteFriend } from "../../../graphql/mutation.js";
-import { NewMatchInvite } from "../../../graphql/subscription.js";
 
-const useStyles = makeStyles((theme) => ({
-	
-}));
+const useStyles = makeStyles((theme) => ({}));
```
Comment (mrrosoff) [inline client/components/App/HeaderBar/Missions.jsx:25]: If we dont use this can we remove?

File: client/components/App/HeaderBar/Missions.jsx [hunk 2]
```diff
@@ -1,55 +1,36 @@
...
-	useEffect(() => {
-		subscribeToMore({
-			document: NewMatchInvite,
-			updateQuery: (prev, { subscriptionData }) => {
-				if (!subscriptionData.data) return prev;
-				return { user: subscriptionData.data.NewMatchInvite };
-			}
-		});
-	}, []);
+	const { loading, error, data } = useQuery(GetHeaderNotifications);
```
Comment (mrrosoff) [inline client/components/App/HeaderBar/Missions.jsx:33]: Can we use a separate query for missions?

File: schema.graphql [hunk 3]
```diff
@@ -146,6 +148,7 @@ type Query {
 	matchLookup(_id: ID!): Match
 	matchSearch(query: String!): [Match!]!
 	shop: Shop!
+	missions: [Mission!]!
```
Comment (mrrosoff) [inline schema.graphql:151]: Why do we have this?

File: server/utils.js [hunk 4]
```diff
@@ -24,17 +28,17 @@ export const checkIsMe = (parent, context) => {
 
 export const checkInMyFriends = async (context, friendUsername) => {
```
Comment (mrrosoff) [inline server/utils.js:29]: This method is async, can we make sure we await all calls?

**Review by mrrosoff — CHANGES_REQUESTED** [review]
Does this work? Will have time to test more thoroughly later. Heres a couple things to fix.

File: schema.graphql (hunk 3 above)
Comment (matthewfernst) [reply schema.graphql:151]: Miss typed but it's gone. Not needed.

File: server/utils.js (hunk 4 above)
Comment (matthewfernst) [reply server/utils.js:29]: updated all with await.

File: client/components/App/HeaderBar/Missions.jsx (hunk 2 above)
Comment (matthewfernst) [reply client/components/App/HeaderBar/Missions.jsx:33]: Is there a downside to using this one?

**Comment by mrrosoff** [comment]
fixes #6

**Review by mrrosoff — APPROVED** [review]

**Pull request merged**